hypershift-operator/controllers/hostedcluster/internal/platform/core.go

```diff
@@ -0,0 +1,60 @@
+package platform
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/upsert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	hostedClusterAnnotation = "hypershift.openshift.io/cluster"
+)
+
+type Platform interface {
+	// ReconcileCAPIInfraCR is responsible for creating or updating a CAPIInfraCR.
```

Comment:
I think we need a little bit more documentation for each of these functions. As it is, I would need to see what the AWS implementation does to know what kind of things I need to do for a new platform.

Reply:
Updated, let me know how it looks now.